test/numerical/TestGemm.cpp

```diff
@@ -61,8 +61,21 @@ static bool isOMGemmTheSameAsNaiveImplFor(const int I, const int J, const int K,
 
   GemmLibBuilder gemm(
       SHARED_LIB_BASE.str(), I, J, K, aTrans, bTrans, cRank, alphaVal, betaVal);
-  return gemm.build() && gemm.compileAndLoad() && gemm.prepareInputs() &&
-         gemm.run() && gemm.verifyOutputs();
+  bool successBuild = gemm.build() && gemm.compileAndLoad();
+  if (!successBuild) {
+    printf("Fail to build.\n");
+    return false;
+  }
+  // Verify generated library
+  std::string instructionName =
+      getenv("TEST_CHECK_INSTRUCTION") ? getenv("TEST_CHECK_INSTRUCTION") : "";
+  std::string sharedLibName =
+      ModelLibBuilder::getSharedLibName(SHARED_LIB_BASE.str());
+  if (!ModelLibBuilder::checkSharedLibInstruction(
+          instructionName, sharedLibName))
+    return false;
+  return successBuild && gemm.prepareInputs() && gemm.run() &&
+         gemm.verifyOutputs();
```

Comment:
use same interface as described above

Reply:
Thanks for your comments. Refactoring done.